hrm-domain/hrm-service/src/profile/sql/profile-list-sql.ts

```diff
@@ -43,17 +43,34 @@ const generateOrderByClause = (clauses: OrderByClauseItem[]): string => {
 };
 
 const selectProfilesUnorderedSql = (whereClause: string) => `
-  SELECT (count(*) OVER())::INTEGER AS "totalCount", id, name
+  SELECT (count(*) OVER())::INTEGER AS "totalCount", *
   FROM "Profiles" profiles
   ${whereClause}
 `;
 
 const listProfilesPaginatedSql = (whereClause: string, orderByClause: string) => `
-  SELECT *
-  FROM (${selectProfilesUnorderedSql(whereClause)}) AS profiles
-  ${orderByClause}
-  OFFSET $<offset>
-  LIMIT $<limit>;
+  WITH TargetProfiles AS (
+    ${selectProfilesUnorderedSql(whereClause)}
+    ${orderByClause}
+    OFFSET $<offset>
+    LIMIT $<limit>
+  ),
+  RelatedProfileFlags AS (
+    SELECT
+      ppf."profileId",
+      JSONB_AGG(ppf."profileFlagId") AS "profileFlags"
```

Comment:
I should have realized this earlier, but I am hoping now to receive the profile flag names instead. So, if it is `name` instead of `profileId`, it will be favourable